huaweicloud/services/css/resource_huaweicloud_css_logstash_cluster.go

```diff
@@ -0,0 +1,547 @@
+	conf := meta.(*config.Config)
+	region := conf.GetRegion(d)
+	cssV1Client, err := conf.HcCssV1Client(region)
+	if err != nil {
+		return diag.Errorf("error creating CSS V1 client: %s", err)
+	}
+
+	createClusterOpts, paramErr := buildlogstashClusterCreateParameters(d, conf)
+	if paramErr != nil {
+		return diag.FromErr(paramErr)
+	}
+
+	r, err := cssV1Client.CreateCluster(createClusterOpts)
+	if err != nil {
+		return diag.Errorf("error creating CSS logstash cluster: %s", err)
+	}
+
+	if (r.Cluster == nil || r.Cluster.Id == nil) && r.OrderId == nil {
+		return diag.Errorf("error creating CSS cluster: id is not found in API response,%#v", r)
+	}
+
+	if r.OrderId == nil {
+		if r.Cluster == nil || r.Cluster.Id == nil {
+			return diag.Errorf("error creating CSS cluster: id is not found in API response,%#v", r)
+		}
+		d.SetId(*r.Cluster.Id)
+	} else {
+		bssClient, err := conf.BssV2Client(conf.GetRegion(d))
+		if err != nil {
+			return diag.Errorf("error creating BSS v2 client: %s", err)
+		}
+
+		// 1. If charging mode is PrePaid, wait for the order to be completed.
+		err = common.WaitOrderComplete(ctx, bssClient, *r.OrderId, d.Timeout(schema.TimeoutCreate))
+		if err != nil {
+			return diag.FromErr(err)
+		}
+
+		// 2. get the resource ID, must be after order success
+		resourceId, err := common.WaitOrderResourceComplete(ctx, bssClient, *r.OrderId, d.Timeout(schema.TimeoutCreate))
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		d.SetId(resourceId)
+	}
+
+	createResultErr := checkClusterCreateResult(ctx, cssV1Client, d.Id(), d.Timeout(schema.TimeoutCreate))
+	if createResultErr != nil {
+		return diag.FromErr(createResultErr)
+	}
+
+	return resourceLogstashClusterRead(ctx, d, meta)
+}
+
+func buildlogstashClusterCreateParameters(d *schema.ResourceData, conf *config.Config) (*model.CreateClusterRequest, error) {
+	createOpts := model.CreateClusterBody{
+		Name: d.Get("name").(string),
+		Datastore: &model.CreateClusterDatastoreBody{
+			Type:    "logstash",
+			Version: d.Get("engine_version").(string),
+		},
+		EnterpriseProjectId: utils.StringIgnoreEmpty(conf.GetEnterpriseProjectID(d)),
+		Tags:                buildLogstashCssTags(d.Get("tags").(map[string]interface{})),
+		Instance: &model.CreateClusterInstanceBody{
+			FlavorRef: d.Get("node_config.0.flavor").(string),
+			Nics: &model.CreateClusterInstanceNicsBody{
+				VpcId:           d.Get("vpc_id").(string),
+				NetId:           d.Get("subnet_id").(string),
+				SecurityGroupId: d.Get("security_group_id").(string),
+			},
+			Volume: &model.CreateClusterInstanceVolumeBody{
+				Size:       int32(d.Get("node_config.0.volume.0.size").(int)),
+				VolumeType: d.Get("node_config.0.volume.0.volume_type").(string),
+			},
+			AvailabilityZone: utils.StringIgnoreEmpty(d.Get("availability_zone").(string)),
+		},
+		InstanceNum: int32(d.Get("node_config.0.instance_number").(int)),
+	}
+
+	if payModel, ok := d.GetOk("period_unit"); ok && d.Get("charging_mode").(string) != "postPaid" {
```

Comment:
```suggestion
	if d.Get("charging_mode").(string) == "prePaid" {
```
only use charging_mode to check